addons/mrp/tests/test_warehouse_multistep_manufacturing.py

```diff
@@ -912,3 +912,40 @@ def test_component_and_byproduct_on_transfers(self):
         self.assertEqual(mo.picking_ids[1].move_ids_without_package[0].product_id, demo)
         self.assertEqual(mo.picking_ids[1].move_ids_without_package[1].product_id, bprod1)
         self.assertEqual(mo.picking_ids[1].move_ids_without_package[2].product_id, bprod2)
+
+    def test_update_mo_from_bom_forecast(self):
+        self.warehouse_1.manufacture_steps = 'pbm_sam'
+        self.env['stock.quant']._update_available_quantity(self.product_2, self.warehouse_1.lot_stock_id, 10)
+        self.env['stock.quant']._update_available_quantity(self.product_3, self.warehouse_1.lot_stock_id, 20)
+        mo = self.env['mrp.production'].create({
+            'bom_id': self.bom_1.id,
+            'picking_type_id': self.warehouse_1.manu_type_id.id,
+        })
+        mo.action_confirm()
+        self.assertEqual(self.product_1.virtual_available, -4)
+        self.assertEqual(self.product_2.virtual_available, 8)
+        rr = self.env['stock.warehouse.orderpoint'].create({
+            'name': 'John Cutter RR',
+            'product_id': self.product_1.id,
+            'warehouse_id': mo.warehouse_id.id,
+        })
+        self.assertEqual(rr.qty_forecast, -4)
+        # Update the BoM
+        self.bom_1.bom_line_ids[1].unlink()
+        self.bom_1.bom_line_ids[0].product_qty = 3
+        self.bom_1.write({
+            'bom_line_ids': [
+                Command.create({'product_id': self.product_3.id, 'product_qty': 5}),
+            ]
+        })
```

Comment:
This works? Feels weird since `bom_1` already has some `bom_line_ids` set to it... Might be safer/cleaner code to just create the mrp.bom.line with a `bom_id` in the vals dict instead?